File: GeneratedEntityReaderClasses/Components/MagicXRayComponent.cs
namespace GeneratedNoitaClasses;
public class MagicXRayComponent : NoitaComponentBase,  iNoitaType<MagicXRayComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt radius { get; set; } = new();
	public NoitaInt steps_per_frame { get; set; } = new();
	public NoitaInt mStep { get; set; } = new();
	public NoitaInt mRadius { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		radius.Read(s);
		steps_per_frame.Read(s);
		mStep.Read(s);
		mRadius.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		radius.Write(s);
		steps_per_frame.Write(s);
		mStep.Write(s);
		mRadius.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/MagicXRayComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class MagicXRayComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    radius:NoitaInt = field(default_factory=NoitaInt)
    steps_per_frame:NoitaInt = field(default_factory=NoitaInt)
    mStep:NoitaInt = field(default_factory=NoitaInt)
    mRadius:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.radius.Read(s)
        self.steps_per_frame.Read(s)
        self.mStep.Read(s)
        self.mRadius.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.radius.Write(s)
        self.steps_per_frame.Write(s)
        self.mStep.Write(s)
        self.mRadius.Write(s)

